master/internal/task/allocation.go

```diff
@@ -475,23 +507,28 @@ func (a *allocation) requestResources() (*sproto.ResourcesSubscription, error) {
 		if err != nil {
 			return nil, errors.Wrap(err, "loading trial allocation")
 		}
-	} else {
-		// Insert new allocation.
-		a.syslog.Debug("requestResources add allocation")
 
-		a.setModelState(model.AllocationStatePending)
-		if err := a.db.AddAllocation(&a.model); err != nil {
-			return nil, errors.Wrap(err, "saving trial allocation")
+		sub, err := a.rm.Allocate(a.req)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to request allocation")
 		}
+		a.sendTaskLog(&model.TaskLog{Log: fmt.Sprintf("Restoring %s (id: %s)", a.req.Name, a.req.AllocationID)})
+		return sub, nil
+	}
+
+	// Insert new allocation.
+	a.syslog.Debug("requestResources add allocation")
+
+	a.setModelState(model.AllocationStatePending)
+	if err := a.db.AddAllocation(&a.model); err != nil {
+		return nil, errors.Wrap(err, "saving trial allocation")
 	}
 
 	sub, err := a.rm.Allocate(a.req)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to request allocation")
 	}
-	a.sendTaskLog(&model.TaskLog{
-		Log: fmt.Sprintf("Scheduling %s (id: %s)", a.req.Name, a.req.AllocationID),
-	})
+	a.sendTaskLog(&model.TaskLog{Log: fmt.Sprintf("Scheduling %s (id: %s)", a.req.Name, a.req.AllocationID)})
 	return sub, nil
```

Comment:
this diff is the same behavior with an extra log, just moved it around for readability.